plugins/woocommerce/includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -796,7 +797,10 @@ protected function update_attributes( &$product, $force = false ) {
 							wp_set_object_terms( $product->get_id(), array(), urldecode( $attribute_key ) );
 						}
 						continue;
-
+					} elseif ( ! is_a( $attribute, 'WC_Product_Attribute' ) ) {
```

Comment:
This method here is run during `create` and `update` in the same class, and both usages are at least currently so that the only way of bailing out would be throwing an error -- i.e., the result isn't checked. Now when we swallow / ignore invalid attributes here the user won't notice and will overwrite the product attributes with these faulty ones removed -- so it is a destructive action. We do log this, but there's no way to recover from it. 

So I suppose my question is: is it OK to just swallow invalid data like this here? Or should we figure out a way to abort the save? Or is there an alternative we could try?

Reply:
(The fix in `find_matching_product_variation` is read-only, so this worry doesn't apply there.)

Reply:
The destructive nature of this can be seen in the final step of the manual testing steps in the PR description.

Reply:
I think we should revert this change, like you mention the risk is high. A buggy 3pd or core code can be fixed in a new version (or by disabling the plugin in a conflict test, for example), but we make the error worse than it was change the problem to data loss from minor inconvinience.